src/common/variant8.cpp

```diff
@@ -0,0 +1,374 @@
+// variant8.c
+
+#include "variant8.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+extern "C" {
+
+variant8_t variant8_init(uint8_t type, uint16_t count, void* pdata)
```

Comment:
the void *pdata will probably force the compiler to assume the parameter must have an address -> crippling optimization possibilities

Reply:
Function variant8_init is used only for pointer types, we need single function which takes pointer to data, for pointer types variant8_init will copy the data using memcpy (fastest way). I do not see better solution.